### PR TITLE
Add new setting: prompt_column_separator

### LINKED ISSUE #483: [FEATURE] Text separator setting
When concatenating individual text columns, we currently use a fixed separator via `separator=""`.

We should add an option to specify the separator via settings.

### PR DESCRIPTION
**Pull request by pascal-pfeiffer**
closes #483

### CODE REVIEW

**Review by sherenem — APPROVED** [review]

**Comment by sherenem** [comment]
Docs LGTM, thank you!

**Review by psinger — CHANGES_REQUESTED** [review]
THX - overall looks good!

A little feedback:
- Shall we add a different default separator, like \n###\n or similar?
- I guess it is not possible to add a nesting that only triggers if multiple values are selected? Possible probably, but some work, not sure if worth.
- Otherwise I would suggest to add it to the tokenizer section though. It is a bit dominant on top.

File: llm_studio/python_configs/text_causal_language_modeling_config.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ class ConfigNLPCausalLMDataset(DefaultConfig):
 
     system_column: str = "system"
     prompt_column: Tuple[str, ...] = ("instruction", "input")
+    prompt_column_separator: str = ""
```
[the remark this answers is not in the document]
Comment (pascal-pfeiffer) [reply llm_studio/python_configs/text_causal_language_modeling_config.py:43]: It was double space before, which is hard to see in the textbox and also a bit unusual. Decided to change it to double newlines now. Will push the changes in a bit. Wdyt?

**Comment by pascal-pfeiffer** [comment]
> * Shall we add a different default separator, like \n###\n or similar?

After some thought, I decided to use `\n\n`, will push the changes in a bit.

> * I guess it is not possible to add a nesting that only triggers if multiple values are selected? Possible probably, but some work, not sure if worth.

I also thought about it, but couldn't find an easy way. It would also force us to use a trigger for the columns dropdown. 

> * Otherwise I would suggest to add it to the tokenizer section though. It is a bit dominant on top.

From a logical standpoint, I believe it should be a dataset setting.

**Review by psinger — APPROVED** [review]

**Pull request merged**